ir/type.h

```diff
@@ -13,6 +13,11 @@ namespace smt { class Model; }
 
 namespace IR {
 
+enum FpType {
```

Comment:
can you move this to the FloatType class pls? Also, don't assign to the enum; doesn't seem necessary?

Reply:
@nunoplopes I don't understand the second sentence. Could you please elaborate a bit?